storage/volumes.md

```diff
@@ -512,12 +518,92 @@ $ docker volume create \
 	--name cif-volume
 ```
 
-Notice the `addr` option is required if using a hostname instead of an IP so docker can perform the hostname lookup.
+The `addr` option is required if you specify a hostname instead of an IP.
+This lets Docker perform the hostname lookup.
+
+### Mount a block storage device
+
+You can mount a block storage device, such as an external drive or a drive partition, to a container.
+The following example shows how to create and use a file as a block storage device,
+and how to mount the block device as a container volume.
+
+> **Important**
+>
+> The following procedure is only an example.
+> The solution illustrated here isn't recommended as a general practice.
+> Don't attempt this approach unless you're very confident about what you're doing.
+{: .important }
+
+Under the hood, the `--mount` flag using the `local` storage driver invokes the
+Linux `mount` syscall and forwards the options you pass to it unaltered.
+Docker doesn't implement any additional functionality on top of the native mount features supported by the Linux kernel.
+
+If you're familiar with the
+[Linux `mount` command](https://man7.org/linux/man-pages/man8/mount.8.html),
+you can think of the `--mount` options being forwarded to the `mount` command as follows:
+
+```console
+$ mount -t <mount.volume-opt.type> <mount.volume-opt.device> <mount.dst> -o <mount.volume-opts.o>
+```
+
+In other words, the following commands are equivalent:
+
+```console
+$ docker run \
+  --mount='type=volume,dst=/external-drive,volume-driver=local,volume-opt=device=/dev/loop5,volume-opt=type=ext4'
+
+$ mount -t ext4 /dev/loop5 /external-drive
+```
+
+The following steps create an `ext4` filesystem and mounts it into a container.
+The filesystem support of your system depends on the version of the Linux kernel you are using.
+
+1. Create a file and allocate some space to it:
+
+   ```console
+   $ fallocate -f 1G disk.raw
+   ```
+
+2. Build a filesystem onto the `disk.raw` file:
+
+   ```console
+   $ mkfs.ext4 disk.raw
+   ```
+
+3. Create a loop device:
+
+   ```console
+   $ losetup -f --show disk.raw
+   /dev/loop5
+   ```
+
+   > **Note**
+   >
+   > `losetup` creates an ephemeral loop device that's removed after
+   > system reboot, or manually removed with `losetup -d`.
+
+4. Run a container that mounts the loop device as a volume:
+
+   ```console
+   $ docker run -it --rm \
+     --mount='type=volume,dst=/external-drive,volume-driver=local,volume-opt=device=/dev/loop5,volume-opt=type=ext4' \
+     ubuntu bash
+   ```
+
+   When the container starts, the path `/external-drive` mounts the
+   `disk.raw` file from the host filesystem as a block device.
+
+5. When you're done, detach the loop device to remove the mount
+   from the host system.
```

Comment:
I'd phrase it more like this, because we didn't mount this device on host:

```suggestion
5. When you're done and the device is unmounted from the container,
   detach the loop device from the host system.
```